lib/models/widget_model.dart

```diff
@@ -23,19 +23,72 @@ class DashboardWidget {
   WidgetConfig config;
   bool disabled;
   Map<String, dynamic> content;
+  DateTime expirationDate;
+  int maxValue;
+  String selectedZabbixMetric;
+  List range;
+  List toDoListItems;
 
   static const WIDGET_STATUS_KEY = 'widgetStatus';
 
-  DashboardWidget({this.id, this.title, this.type, this.config, this.disabled, this.content});
+  DashboardWidget({
+    this.id,
+    this.title,
+    this.type,
+    this.config,
+    this.disabled,
+    this.content,
+    this.expirationDate,
+    this.maxValue,
+    this.selectedZabbixMetric,
+    this.range,
+    this.toDoListItems,
+  });
 
   factory DashboardWidget.fromJson(Map<String, dynamic> json) => DashboardWidget(
       id: json['id'],
       title: json['title'],
       type: json['type'],
       config: WidgetConfig.fromJson(json['config']),
       disabled: json['disabled'],
+      expirationDate: (json['expirationDate'] != "null" && json['expirationDate'] != null)
```

Comment:
this is strange why we need to chceck against string `null` and object `null`?

Reply:
Because in some cases I don't remebere which maybe from WS, the expiration date was not null but a string with "null"  so the result of assertion json['expirationDate'] != null was true and we had error when parsing DateTime.parse(json['expirationDate'])

Reply:
In this case it looks like this is a fix for a bug in wrong place.

Reply:
will be changed in change-request-part1 PR